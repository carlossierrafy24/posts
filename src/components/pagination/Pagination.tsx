import React from 'react';
import Page from '../page/Page';
import { Post, Comment, User } from '../../data/Interfaces';

const { REACT_APP_POSTS_API, REACT_APP_COMMENTS_API, REACT_APP_USERS_API } = process.env;

const Pagination: React.FC = (): JSX.Element => {
    const [posts, setPosts] = React.useState<Post[]>([]);
    const [comments, setComments] = React.useState<Comment[]>([]);
    const [users, setUsers] = React.useState<User[]>([]);
    const [currentPage, setCurrentPage] = React.useState<number>(1);

    // array of numbers from 1 to 10 for pages indicator
    const pageNumbers = [];
    for (let i = 1; i <= 10; i++) {
        pageNumbers.push(i);
    }

    React.useEffect(() => {
        fetch(`${REACT_APP_POSTS_API}`)
            .then((res) => res.json())
            .then((data) => setPosts(data))
            .catch((err) => console.log(err));
        fetch(`${REACT_APP_COMMENTS_API}`)
            .then((res) => res.json())
            .then((data) => setComments(data))
            .catch((err) => console.log(err));
        fetch(`${REACT_APP_USERS_API}`)
            .then((res) => res.json())
            .then((data) => setUsers(data))
            .catch((err) => console.log(err));
    }, []);

    const handleNextPage = () => {
        currentPage < 10 && setCurrentPage(currentPage + 1);
    };

    const handlePreviousPage = () => {
        currentPage > 1 && setCurrentPage(currentPage - 1);
    };

    return (
        <div>
            <Page post={posts.slice(currentPage === 1 ? 0 : currentPage * 10 - 10, currentPage === 1 ? 10 : currentPage * 10)} comment={comments} users={users} />
            <div className="pagination">
                <button className="navigation-btn" onClick={handlePreviousPage}>
                    Prev
                </button>
                <div>
                    {pageNumbers.map((number) => (
                        <button key={number} onClick={() => setCurrentPage(number)} className={currentPage === number ? 'active-page' : 'inactive-page'}>
                            {number}
                        </button>
                    ))}
                </div>
                <button className="navigation-btn" onClick={handleNextPage}>
                    Next
                </button>
            </div>
        </div>
    );
};
export default Pagination;
