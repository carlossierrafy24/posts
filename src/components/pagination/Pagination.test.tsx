import React from 'react';
import Pagination from './Pagination';
import { render, screen } from '@testing-library/react';

test('should render pagination buttons', () => {
    render(<Pagination />);
    expect(screen.getByText('Prev')).toBeInTheDocument();
    expect(screen.getByText('Next')).toBeInTheDocument();
});
test('should render pagination buttons disabled', () => {
    render(<Pagination />);
    expect(screen.getByText('Prev').classList.contains('disabled')).toBe(true);
});
