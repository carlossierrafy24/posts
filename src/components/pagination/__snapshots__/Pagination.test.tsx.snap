// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render pagination 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="grid-post-container-scroll mt-5 scroll-m-2 overflow-y-auto"
        >
          <div
            class="grid-post-container"
          />
        </div>
        <div
          class="pagination"
        >
          <button
            class="navigation-btn"
          >
            Prev
          </button>
          <div>
            <button
              class="active-page"
            >
              1
            </button>
            <button
              class="inactive-page"
            >
              2
            </button>
            <button
              class="inactive-page"
            >
              3
            </button>
            <button
              class="inactive-page"
            >
              4
            </button>
            <button
              class="inactive-page"
            >
              5
            </button>
            <button
              class="inactive-page"
            >
              6
            </button>
            <button
              class="inactive-page"
            >
              7
            </button>
            <button
              class="inactive-page"
            >
              8
            </button>
            <button
              class="inactive-page"
            >
              9
            </button>
            <button
              class="inactive-page"
            >
              10
            </button>
          </div>
          <button
            class="navigation-btn"
          >
            Next
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="grid-post-container-scroll mt-5 scroll-m-2 overflow-y-auto"
      >
        <div
          class="grid-post-container"
        />
      </div>
      <div
        class="pagination"
      >
        <button
          class="navigation-btn"
        >
          Prev
        </button>
        <div>
          <button
            class="active-page"
          >
            1
          </button>
          <button
            class="inactive-page"
          >
            2
          </button>
          <button
            class="inactive-page"
          >
            3
          </button>
          <button
            class="inactive-page"
          >
            4
          </button>
          <button
            class="inactive-page"
          >
            5
          </button>
          <button
            class="inactive-page"
          >
            6
          </button>
          <button
            class="inactive-page"
          >
            7
          </button>
          <button
            class="inactive-page"
          >
            8
          </button>
          <button
            class="inactive-page"
          >
            9
          </button>
          <button
            class="inactive-page"
          >
            10
          </button>
        </div>
        <button
          class="navigation-btn"
        >
          Next
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
